### PR TITLE
Make mathematica_test a little less brittle

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: numerics/polynomial.hpp [hunk 1]
```diff
@@ -31,12 +31,12 @@ FORWARD_DECLARE_FUNCTION_FROM(
     mathematica,
     TEMPLATE(typename Value, typename Argument, int degree_,
              template<typename, typename, int> typename Evaluator,
-             typename OptionalExpressIn = std::nullopt_t) std::string,
+             typename OptionalExpressIn) std::string,
```
Comment (pleroy) [inline numerics/polynomial.hpp:34]: This is fishy.  Similar declarations exist in `poisson_series.hpp` and `piecewise_poisson_series.hpp`, so I presume that they would need changing too.  As a matter of fact, it's seems better to have the default live in `mathematica` than in distant units that happen to have a forward declaration.

Comment (eggrobin) [reply numerics/polynomial.hpp:34]: > Similar declarations exist in `poisson_series.hpp` and `piecewise_poisson_series.hpp`

They don’t (they don’t have defaults).

> As a matter of fact, it's seems better to have the default live in `mathematica` than in distant units that happen to have a forward declaration.

This is what this pull request does.

File: mathematica/mathematica_test.cpp [hunk 2]
```diff
@@ -106,170 +100,97 @@ TEST_F(MathematicaTest, ToMathematica) {
...
     EXPECT_EQ("List[]", ToMathematica(FixedVector<int, 0>()));
   }
   {
-    EXPECT_EQ(
-        "List["
-        "SetPrecision[+2.00000000000000000*^+00,$MachinePrecision],"
-        "SetPrecision[+3.00000000000000000*^+00,$MachinePrecision],"
-        "SetPrecision[-4.00000000000000000*^+00,$MachinePrecision]]",
+    EXPECT_EQ("List[" + ToMathematica(2.0) + "," + ToMathematica(3.0) + "," +
+                  ToMathematica(-4.0) + "]",
```
[the remark this answers is not in the document]
Comment (eggrobin) [reply mathematica/mathematica_test.cpp:109]: That workflow seems backward; you should generally be able to write the test before you first run the code.

File: mathematica/mathematica_test.cpp [hunk 3]
```diff
@@ -561,33 +330,16 @@ TEST_F(MathematicaTest, ExpressIn) {
...
-        "SetPrecision[+3.00000000000000000*^+00,$MachinePrecision],"
-        "SetPrecision[-4.00000000000000000*^+00,$MachinePrecision]],"
-        "List["
-        "SetPrecision[-1.00000000000000000*^+00,$MachinePrecision],"
-        "SetPrecision[-5.00000000000000000*^+00,$MachinePrecision],"
-        "SetPrecision[+8.00000000000000000*^+00,$MachinePrecision]]]]",
-        ToMathematica(*trajectory.begin(), ExpressIn(Metre, Second)));
+        ToMathematica(std::tuple{0.0,
+                                 std::tuple{std::tuple{2.0, 3.0, -4.0},
+                                            std::tuple{-1.0, -5.0, 8.0}}}),
```
[the remark this answers is not in the document]
Comment (eggrobin) [reply mathematica/mathematica_test.cpp:327]: As discussed, ended up adding more tuples instead.

**Comment by eggrobin** [comment]
retest this please

**Pull request merged**